deepmedic/frontEnd/configParsing/trainSessionParams.py

```diff
@@ -7,346 +7,455 @@
-        print("ERROR: The parameter \"classicMom0OrNesterov1\" must be given 0 or 1. Omit for default. Exiting!"); exit(1)
+    def errorRequireMomentumClass0Nestov1():
+        print("ERROR: The parameter \"classicMom0OrNesterov1\" must be given 0 or 1. Omit for default. Exiting!");
+        exit(1)
+
     @staticmethod
-    def errorRequireMomValueBetween01() :
-        print("ERROR: The parameter \"momentumValue\" must be given between 0.0 and 1.0 Omit for default. Exiting!"); exit(1)
+    def errorRequireMomValueBetween01():
+        print("ERROR: The parameter \"momentumValue\" must be given between 0.0 and 1.0 Omit for default. Exiting!");
+        exit(1)
+
     @staticmethod
-    def errorRequireMomNonNorm0Norm1() :
-        print("ERROR: The parameter \"momNonNorm0orNormalized1\" must be given 0 or 1. Omit for default. Exiting!"); exit(1)
-    
+    def errorRequireMomNonNorm0Norm1():
+        print("ERROR: The parameter \"momNonNorm0orNormalized1\" must be given 0 or 1. Omit for default. Exiting!");
+        exit(1)
+
     def __init__(self,
-                log,
-                mainOutputAbsFolder,
-                folderForSessionCnnModels,
-                folderForPredictionsVal,
-                folderForFeaturesVal,
-                num_classes,
-                model_name,
-                cfg):
-        
-        #Importants for running session.
+                 log,
+                 mainOutputAbsFolder,
+                 folderForSessionCnnModels,
+                 folderForPredictionsVal,
+                 folderForFeaturesVal,
+                 num_classes,
+                 model_name,
+                 cfg):
+
+        # Importants for running session.
         # From Session:
         self.log = log
         self.mainOutputAbsFolder = mainOutputAbsFolder
-        
+
         # From Config:
-        self.sessionName = self.getSessionName( cfg[cfg.SESSION_NAME] )
-        
+        self.sessionName = self.getSessionName(cfg[cfg.SESSION_NAME])
+
         abs_path_to_cfg = cfg.get_abs_path_to_cfg()
-        abs_path_to_saved = getAbsPathEvenIfRelativeIsGiven( cfg[cfg.SAVED_MODEL], abs_path_to_cfg ) if cfg[cfg.SAVED_MODEL] is not None else None # Load pretrained model.
-        self.savedModelFilepath = check_and_adjust_path_to_ckpt( self.log, abs_path_to_saved) if abs_path_to_saved is not None else None
-        
-        #====================TRAINING==========================
+        abs_path_to_saved = getAbsPathEvenIfRelativeIsGiven(cfg[cfg.SAVED_MODEL], abs_path_to_cfg) if cfg[
+                                                                                                          cfg.SAVED_MODEL] is not None else None  # Load pretrained model.
+        self.savedModelFilepath = check_and_adjust_path_to_ckpt(self.log,
+                                                                abs_path_to_saved) if abs_path_to_saved is not None else None
+
+        self.tensorboardLog = bool(cfg[cfg.TENSORBOARD_LOG])
```

Comment:
Very concise, but allows values outside {0,1}. E.g. bool(99) is True. I would either make them give either True/False/None, or here check for arbitrary values and raise errors. Probably True/False is clearer.

Reply:
I remember that at the time I had issues with boolean inputs, and that's why I used the {1,0}. I agree True/False is better